server/controllers/notifications.controller.js

```diff
@@ -21,16 +23,48 @@ function sendPushNotifications(req, res, next) { // eslint-disable-line no-unuse
             },
         },
     }).then((users) => {
-        if (!users) return;
+        if (!users.length > 0) {
+          res.status(404);
+          res.send({error: "Users Not Found"});
+          return
+        }
         users.forEach((user) => {
             if (!user) return;
-            const userData = pushData.find(_userData => _userData.username === user.username);
-            pushNotificationHelper.sendPushNotification(user, userData);
+            const userData = data.find(_userData => _userData.username === user.username);
+            switch (protocol) {
+            case 'push':
+                pushNotificationHelper.sendPushNotification(user, userData);
```

Comment:
Structure of userData differs depending on Helper, with no validation in the helper functions.

pushNotificationHelper, [iosPushData](https://github.com/amida-tech/amida-notification-microservice/blob/da1167ccd5cb35d3d596cceabe43a93057873a96/server/helpers/pushNotificationHelper.js#L75)
- body.body
- data.title (presumably same as subject)
- body.topic

[sendEmail](https://github.com/amida-tech/amida-notification-microservice/blob/da1167ccd5cb35d3d596cceabe43a93057873a96/server/helpers/emailHelper.js#L5)
- data.body (email body)
- data.email
- data.subject

[sendSms](https://github.com/amida-tech/amida-notification-microservice/blob/da1167ccd5cb35d3d596cceabe43a93057873a96/server/helpers/smsHelper.js)
- data.message
- data.phone
- data.subject

Reply:
Overall payload structure will however be similar across all protocols. As of now, it's sent in body.data and will be an array of objects with each object containing a username.